### PR TITLE
#190 add update datasource connection

### PR DESCRIPTION
**Pull request by RussTheAerialist**
Clean up the API a bit because I didn't really click with it.

### CODE REVIEW

File: samples/update_connection.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import argparse
+import getpass
+import logging
+
+import tableauserverclient as TSC
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Get all of the refresh tasks available on a server')
```
Comment (t8y8) [inline samples/update_connection.py]: The description is for the wrong sample (I bet a lot of our samples are, actually)

File: samples/update_connection.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        endpoint = {
+            'workbook': server.workbooks,
+            'datasource': server.datasources
+        }.get(args.resource_type)
+
+        update_function = endpoint.update_connection
+        resource = endpoint.get_by_id(args.resource_id)
+        endpoint.populate_connections(resource)
+        connections = filter(lambda x: x.id == args.connection_id, resource.connections)
+        assert(len(connections) == 1)
```
Comment (t8y8) [inline samples/update_connection.py]: Do you have to `list(filter` Here for python3? I forget if filter_results objects have len

File: tableauserverclient/models/connection_item.py [hunk 3]
```diff
@@ -29,6 +29,10 @@ def id(self):
     def connection_type(self):
         return self._connection_type
 
+    def __repr__(self):
+        return "<ConnectionItem#{_id} embed={embed_password} type={_connection_type} username={username}>"\
```
Comment (t8y8) [inline tableauserverclient/models/connection_item.py]: One day we can use f-strings. One day...

File: tableauserverclient/server/endpoint/workbooks_endpoint.py [hunk 4]
```diff
@@ -87,11 +87,16 @@ def update(self, workbook_item):
         return updated_workbook._parse_common_tags(server_response.content, self.parent_srv.namespace)
 
     # Update workbook_connection
-    def update_conn(self, workbook_item, connection_item):
+    @api(version="2.3")
+    def update_connection(self, workbook_item, connection_item):
```
Comment (t8y8) [inline tableauserverclient/server/endpoint/workbooks_endpoint.py]: Should probably set `update_conn` to `update_connection` for back compact; and throw a warning?

**Comment by t8y8** [comment]
I like it!

We may want to let the guy who was working on adding multiple connections know (did that ever get done? I should pester Ben...)

File: samples/update_connection.py (hunk 2 above)
Comment (RussTheAerialist) [reply samples/update_connection.py]: Doh. Yes.

File: tableauserverclient/server/endpoint/workbooks_endpoint.py (hunk 4 above)
Comment (RussTheAerialist) [reply tableauserverclient/server/endpoint/workbooks_endpoint.py]: Probably... I'll make the change.

**Comment by bheem123** [comment]
I tested the update_connection.py on my server version 2018.2, and it does not work. No error,  but nothing is updated on server. Did this work for anyone?

**Pull request merged**